models/agent_helpers.go

```diff
@@ -181,6 +181,12 @@ func AgentAddExporter(q *reform.Querier, agentType AgentType, params *AddExporte
 		return nil, err
 	}
 
+	if params.PMMAgentID != "" {
```

Comment:
I think it's wrong, isn't it? If there is no pmm-agent id, you can't add exporter at all, as all exporters must be connected to pmm-agent. So this function should return an error when this parameter is empty.

Reply:
As I know there might be external exporter with no pmm-agent id

Reply:
Yeah, but we don't have it now. Also, the logic of it (how it should be created) can be different. I think we need separate function for adding external_exporter as we have a different function `AgentAddPmmAgent()`. But let's ask @AlekSi what does he think.

Reply:
When I was refactoring this code, first of all, I create different functions for each exporter and only after that I made     `AgentAddExporter` when I have realized that they all work the same way. As of external_exporter, I think it's better to implement that and then look is it different and can we refactor it.

Reply:
Yeah, you are right

Reply:
> I think we need separate function for adding external_exporter as we have a different function AgentAddPmmAgent()

+1
But I also think that `AgentAddExporter` is not a great name – QAN data providers (built-in Agents for performance schema and slow log) are not exporters, and they are not "special" like pmm-agent or external exporters. Anyway, let's not change that now.

Speaking of external exporters, I think we will have them in pmm-managed (database, models, services, Inventory APIs) in 2.0 _mostly_ to get the data model right. We will expose them in Management API (if at all) and in `pmm-admin` in 2.x.